drivers/sensor/mpu9250/mpu9250.c

```diff
@@ -0,0 +1,350 @@
+		val->val2 += 1000000;
+	} else if (val->val2 >= 1000000) {
+		val->val1++;
+		val->val2 -= 1000000;
+	}
+}
+
+static int mpu9250_channel_get(const struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct mpu9250_data *drv_data = dev->data;
+#ifdef CONFIG_MPU9250_MAGN_EN
+	int ret;
+#endif
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_XYZ:
+		mpu9250_convert_accel(val, drv_data->accel_x,
+				      drv_data->accel_sensitivity_shift);
+		mpu9250_convert_accel(val + 1, drv_data->accel_y,
+				      drv_data->accel_sensitivity_shift);
+		mpu9250_convert_accel(val + 2, drv_data->accel_z,
+				      drv_data->accel_sensitivity_shift);
+		break;
+	case SENSOR_CHAN_ACCEL_X:
+		mpu9250_convert_accel(val, drv_data->accel_x,
+				      drv_data->accel_sensitivity_shift);
+		break;
+	case SENSOR_CHAN_ACCEL_Y:
+		mpu9250_convert_accel(val, drv_data->accel_y,
+				      drv_data->accel_sensitivity_shift);
+		break;
+	case SENSOR_CHAN_ACCEL_Z:
+		mpu9250_convert_accel(val, drv_data->accel_z,
+				      drv_data->accel_sensitivity_shift);
+		break;
+	case SENSOR_CHAN_GYRO_XYZ:
+		mpu9250_convert_gyro(val, drv_data->gyro_x,
+				     drv_data->gyro_sensitivity_x10);
+		mpu9250_convert_gyro(val + 1, drv_data->gyro_y,
+				     drv_data->gyro_sensitivity_x10);
+		mpu9250_convert_gyro(val + 2, drv_data->gyro_z,
+				     drv_data->gyro_sensitivity_x10);
+		break;
+	case SENSOR_CHAN_GYRO_X:
+		mpu9250_convert_gyro(val, drv_data->gyro_x,
+				     drv_data->gyro_sensitivity_x10);
+		break;
+	case SENSOR_CHAN_GYRO_Y:
+		mpu9250_convert_gyro(val, drv_data->gyro_y,
+				     drv_data->gyro_sensitivity_x10);
+		break;
+	case SENSOR_CHAN_GYRO_Z:
+		mpu9250_convert_gyro(val, drv_data->gyro_z,
+				     drv_data->gyro_sensitivity_x10);
+		break;
+#ifdef CONFIG_MPU9250_MAGN_EN
+	case SENSOR_CHAN_MAGN_XYZ:
+		ret = ak8963_convert_magn(val, drv_data->magn_x,
+					  drv_data->magn_scale_x,
+					  drv_data->magn_st2);
+		if (ret < 0) {
+			return ret;
+		}
+		ret = ak8963_convert_magn(val + 1, drv_data->magn_y,
+					  drv_data->magn_scale_y,
+					  drv_data->magn_st2);
+		if (ret < 0) {
+			return ret;
+		}
+		ret = ak8963_convert_magn(val + 2, drv_data->magn_z,
+					  drv_data->magn_scale_z,
+					  drv_data->magn_st2);
+		return ret;
+	case SENSOR_CHAN_MAGN_X:
+		return ak8963_convert_magn(val, drv_data->magn_x,
+				    drv_data->magn_scale_x,
+				    drv_data->magn_st2);
+	case SENSOR_CHAN_MAGN_Y:
+		return ak8963_convert_magn(val, drv_data->magn_y,
+				    drv_data->magn_scale_y,
+				    drv_data->magn_st2);
+	case SENSOR_CHAN_MAGN_Z:
+		return ak8963_convert_magn(val, drv_data->magn_z,
+				    drv_data->magn_scale_z,
+				    drv_data->magn_st2);
+	case SENSOR_CHAN_DIE_TEMP:
+		mpu9250_convert_temp(val, drv_data->temp);
+		break;
+#endif
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+#ifdef CONFIG_MPU9250_MAGN_EN
+#define MPU9250_READ_BUF_SIZE 11
+#else
+#define MPU9250_READ_BUF_SIZE 7
+#endif
```

Comment:
nit: better move this to the top of the file

Reply:
Moved